wzdx/combination/attenuator.py

```diff
@@ -0,0 +1,190 @@
+import json
+from datetime import datetime
+from wzdx.tools import cdot_geospatial_api, polygon_tools
+import logging
+
+ATTENUATOR_TIME_AHEAD_SECONDS = 30 * 60
+ISO_8601_FORMAT_STRING = "%Y-%m-%dT%H:%M:%SZ"
+
+
+def get_route_info_planned_event(planned_event):
+    start = cdot_geospatial_api.get_route_and_measure(
+        planned_event['geometry']['coordinates'][0])
+    end = cdot_geospatial_api.get_route_and_measure(
+        planned_event['geometry']['coordinates'][1])
+    if start['Route'] != end['Route'] or start['Measure'] == end['Measure']:
+        return None
+    else:
+        return {
+            'route': start['Route'],
+            'start_measure': start['Measure'],
+            'end_measure': end['Measure']
+        }
+
+
+def get_route_info_geotab(geotab):
+    start = cdot_geospatial_api.get_route_and_measure(
+        geotab['geometry']['coordinates'][0])
+    end = cdot_geospatial_api.get_route_and_measure(
+        geotab['geometry']['coordinates'][1])
+    if start['Route'] != end['Route'] or start['Measure'] == end['Measure']:
+        return None
+    else:
+        return {
+            'route': start['Route'],
+            'start_measure': start['Measure'],
+            'end_measure': end['Measure']
+        }
+
+
+def json_serial(obj):
+    """JSON serializer for objects not serializable by default json code"""
+
+    if isinstance(obj, (datetime)):
+        return obj.strftime(ISO_8601_FORMAT_STRING)
+    raise TypeError("Type %s not serializable" % type(obj))
+
+
+def main():
+    with open('./wzdx/sample_files/raw/geotab_avl/geotab_all.json') as f:
+        geotab_avl = json.loads(f.read())
+    with open('./wzdx/sample_files/enhanced/planned_events/planned_event_all.json') as f:
+        planned_event = json.loads(f.read())
+
+    combined_events = get_combined_events(geotab_avl, planned_event)
+
+    with open('./wzdx/sample_files/enhanced/planned_events/planned_event_combined.json', 'w+') as f:
+        f.write(json.dumps(combined_events, indent=2))
+
+
+def validate_directionality(geotab, planned_event):
+    geotab_bearing = geotab['avl_location']['position']['bearing']
+    planned_event_direction = planned_event['features'][0]['properties']['core_details']['direction']
+
+    geotab_direction = polygon_tools.get_closest_direction_from_bearing(
+        geotab_bearing, planned_event_direction)
+
+    print(geotab_direction, planned_event_direction)
+
+    return geotab_direction == planned_event_direction
+
+
+def get_combined_events(gebtab_msgs, planned_events):
+    conmbined_events = []
+    for i in identify_overlapping_features(gebtab_msgs, planned_events):
+        if not validate_directionality(*i):
```

Comment:
This asterisk iteration syntax threw me for a bit, had to look it up. Interesting way for handling the case!